File: leetcode_problems/typescript/src/lc2696_min_string_len_after_removing_substrings.ts
function minLength(s: string): number {
  let stack: string[] = [];
  for (let char of s) {
    if (stack.length > 0 && char === "D" && stack[stack.length - 1] === "C") {
      stack.pop();
    } else if (
      stack.length > 0 &&
      char === "B" &&
      stack[stack.length - 1] === "A"
    ) {
      stack.pop();
    } else {
      stack.push(char);
    }
  }
  return stack.length;
}

File: leetcode_problems/cpp/lc2696_min_string_len_after_removing_substrings.cpp
#include <string>
#include <stack>
using std::stack;
using std::string;

class Solution
{
public:
    int minLength(string s)
    {
        stack<char> st;
        for (char chr : s)
        {
            if (!st.empty() && chr == 'B' && st.top() == 'A')
            {
                st.pop();
            }
            else if (!st.empty() && chr == 'D' && st.top() == 'C')
            {
                st.pop();
            }
            else
            {
                st.push(chr);
            }
        }
        return st.size();
    }
};